apps/job-service/src/jobs/delayed-jobs.controller.ts

```diff
@@ -1,38 +1,116 @@
+  }
+
   @Get(':uuid')
   @ApiOperation({
     operationId: 'delayedJobsFind',
-    description: 'Get the current status and potentially payload or error from a delayed job.',
+    description:
+      'Get the current status and potentially payload or error from a delayed job.',
   })
   @JsonApiResponse({ data: { type: DelayedJobDto } })
   @ApiException(() => UnauthorizedException, {
     description: 'Authentication failed.',
   })
   @ApiException(() => NotFoundException, {
-    description: 'Job with that UUID not found.'
+    description: 'Job with that UUID not found.',
   })
+  // Note: Since jobs are very generic and we don't track which resources are related to a given
+  // job, there is no effective way to make a policy for jobs until we expand the service to
+  // include an owner ID on the job table.
   async findOne(@Param('uuid') pathUUID: string): Promise<JsonApiDocument> {
-    const job = await DelayedJob.findOne({ where: { uuid: pathUUID }});
+    const job = await DelayedJob.findOne({ where: { uuid: pathUUID } });
     if (job == null) throw new NotFoundException();
 
-    // Note: Since jobs are very generic and we don't track which resources are related to a given
-    // job, there is no effective way to make a policy for jobs until we expand the service to
-    // include an owner ID on the job table.
-
     return buildJsonApi()
       .addData(pathUUID, new DelayedJobDto(job))
       .document.serialize();
   }
-}
+
+  @Patch('bulk-clear')
+  @ApiOperation({
+    operationId: 'bulkClearJobs',
+    summary: 'Bulk update jobs to modify isAcknowledged for specified job IDs',
+    description: `Accepts a JSON:API-compliant payload to bulk update jobs, allowing each job's isAcknowledged attribute to be set to true or false.`,
+  })
+  @JsonApiResponse({ data: { type: DelayedJobDto } })
+  @ApiException(() => UnauthorizedException, { description: 'Authentication failed.' })
+  @ApiException(() => BadRequestException, { description: 'Invalid payload or IDs provided.' })
+  @ApiException(() => NotFoundException, { description: 'One or more jobs specified in the payload could not be found.' })
+  async bulkClearJobs(
+    @Body() bulkClearJobsDto: DelayedJobBulkUpdateBodyDto,
+    @Request() { authenticatedUserId }
+  ): Promise<JsonApiDocument> {
+    const jobUpdates = bulkClearJobsDto.data;
+    const updatePromises = jobUpdates.map(async (job) => {
+      const [updatedCount] = await DelayedJob.update(
+        { isAcknowledged: job.attributes.isAcknowledged },
+        {
+          where: {
+            uuid: job.uuid,
+            createdBy: authenticatedUserId,
+            status: { [Op.ne]: 'pending' },
+          },
+        });
+    
+      if (updatedCount === 0) {
+        throw new NotFoundException(`Job with UUID ${job.uuid} could not be updated.`);
+      }
+    
+      const updatedJob = await DelayedJob.findOne({
+        where: { uuid: job.uuid },
+      });
+
+      return updatedJob;
+    });
```

Comment:
I see how you got here, but it's a little weird to update and then fetch. Let's change that around, reducing the number of queries in the process, and preventing a partial update if some of the jobs aren't found:
```
const jobs = await DelayedJob.findAll({ 
  where: {
    uuid: { [Op.in]: jobUpdates.map(({ uuid }) => uuid) },
    createdBy: authenticatedUserId,
    status: { [Op.ne]: 'pending' }
  }
});
if (jobs.length !== jobUpdates.length) {
  throw new NotFoundException('Some jobs in the request could not be updated');
}

const updatePromises = jobUpdates.map(async ({ uuid, attributes }) => {
  const job = jobs.find((job) => job.uuid === uuid);
  job.isAcknowledged = attributes.isAcknowledged;
  await job.save();
  return job;
});
```